pkg/kds/mux/zone_watch.go

```diff
@@ -110,11 +115,69 @@ func (zw *ZoneWatch) Start(stop <-chan struct{}) error {
 						TenantID: zone.tenantID,
 					})
 					delete(zw.zones, zone)
+					delete(zw.zoneStreams, zone)
+				}
+				// Now we want to check individual stream
+				for stream, connOpenTime := range zw.zoneStreams[zone] {
+					var conf *system_proto.KDSStream
+					switch stream {
+					case service.Clusters:
+						conf = zoneInsight.Spec.GetKdsStreams().GetClusters()
```

Comment:
nit: you can probably create helper func in `zone_insight_helper.go`

```go
func (x *ZoneInsight) GetKDSStream(streamType) *KDSStream 
```

and use it here and in `service.go`